### PR TITLE
Switch to eslint

### PR DESCRIPTION
**Pull request by andriichumak**
Since TSLint is now deprecated, switching to ESLint with TSLint plugin. A few rules had to be adjusted or temporary disabled.

ESLint is setup to treat the whole repo as a single instance. No individual commands to run linter on each package.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    },
+    'plugins': [
+        '@typescript-eslint',
+        '@typescript-eslint/tslint'
+    ],
+    'rules': {
+        '@typescript-eslint/adjacent-overload-signatures': 'error',
+        '@typescript-eslint/array-type': ['error', {'default': 'array-simple'}],
+        '@typescript-eslint/ban-types': 'error',
+        '@typescript-eslint/class-name-casing': 'error',
```
Comment (JeKachan) [inline .eslintrc.js]: cant find such rule in
https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        'sourceType': 'module'
+    },
+    'plugins': [
+        '@typescript-eslint',
+        '@typescript-eslint/tslint'
+    ],
+    'rules': {
+        '@typescript-eslint/adjacent-overload-signatures': 'error',
+        '@typescript-eslint/array-type': ['error', {'default': 'array-simple'}],
+        '@typescript-eslint/ban-types': 'error',
```
Comment (JeKachan) [inline .eslintrc.js]: here should be an array according to docs.

File: .eslintrc.js (hunk 1 above)
Comment (andriichumak) [reply .eslintrc.js]: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/class-name-casing.md

File: .eslintrc.js (hunk 2 above)
Comment (andriichumak) [reply .eslintrc.js]: With a string 'error' it's using default settings. Honestly, I did not set up each rule individually. There is a tool that converts TSLint config into ESLint config - I used it and only adjusted the rules that caused problems with our code styling.

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        '@typescript-eslint/class-name-casing': 'error',
+        '@typescript-eslint/consistent-type-assertions': 'error',
+        '@typescript-eslint/consistent-type-definitions': 'off',
+        '@typescript-eslint/explicit-member-accessibility': [
```
Comment (JeKachan) [inline .eslintrc.js]: imho, this is not usefull for us, just will add more boilerplate.

Comment (andriichumak) [reply .eslintrc.js]: Agree, will drop.

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+            }
+        ],
+        '@typescript-eslint/member-ordering': 'error',
+        '@typescript-eslint/no-empty-function': 'error',
+        '@typescript-eslint/no-empty-interface': 'error',
+        '@typescript-eslint/no-explicit-any': 'off',
+        '@typescript-eslint/no-misused-new': 'error',
+        '@typescript-eslint/no-namespace': 'off',
+        '@typescript-eslint/no-parameter-properties': 'off',
+		'@typescript-eslint/no-unused-vars': ['error', {'varsIgnorePattern': 'jsx'}],
```
Comment (JeKachan) [inline .eslintrc.js]: redundant tab character

**Comment by andriichumak** [comment]
Merging this one, will add another PR if other changes are needed.

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+			},
+		],
+		'@typescript-eslint/semi': [
+			'error',
+			'always',
+		],
+		'@typescript-eslint/triple-slash-reference': 'error',
+		'@typescript-eslint/type-annotation-spacing': 'error',
+		'@typescript-eslint/unified-signatures': 'error',
+		'arrow-body-style': 'error',
```
Comment (JeKachan) [inline .eslintrc.js]: I would disable this rule

**Pull request merged**